lib/portage/gpg.py

```diff
@@ -22,6 +23,7 @@ def __init__(self, settings):
         """
         Portage settings are needed to run GPG unlock command.
         """
+        self.GPG_tmp_file = "/tmp/portage_gpg_output"
```

Comment:
Using a fixed file name in /tmp is a really bad idea, especially for a program that often runs as root.

I would suggest using [NamedTemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) to securely create a unique filename on every invocation of `unlock()`.